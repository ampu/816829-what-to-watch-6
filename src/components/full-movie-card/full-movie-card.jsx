import React from 'react';
import {Link, Redirect, generatePath} from 'react-router-dom';
import {connect} from 'react-redux';

import {MainPath} from '../../constants/paths';
import PosterSize from '../../constants/poster-size';
import OperationStatus from '../../constants/operation-status';

import {selectMovieById, selectMoviesStatus} from '../../store/selectors/movie-selectors';
import {selectLoginStatus} from '../../store/selectors/user-selectors';
import {selectAlikeMovies} from '../../store/selectors/genre-selectors';

import Logo from '../logo/logo';
import MoviesList, {MoviesList as OriginalMoviesList} from '../movies-list/movies-list';
import UserBlock from '../user-block/user-block';
import MovieDescription from '../movie-description/movie-description';
import MovieBackground from '../movie-background/movie-background';
import MovieInfo from './movie-info';
import SpinnerLoading from '../spinner-loading/spinner-loading';
import Maintenance from '../maintenance/maintenance';
import Container from '../container/container';
import Footer from '../footer/footer';

import operationStatusType from '../../typings/operation-status-type';
import {movieShape} from '../../typings/movie-type';

const FullMovieCard = ({loginStatus, moviesStatus, movie = {}, alikeMovies = []}) => {

  if (moviesStatus === OperationStatus.PENDING) {
    return (
      <Container isCentered>
        <SpinnerLoading/>
      </Container>
    );
  }

  if (moviesStatus === OperationStatus.REJECTED) {
    return (
      <Container isCentered>
        <Maintenance/>
      </Container>
    );
  }

  if (!movie.id) {
    return <Redirect to={MainPath.NOT_FOUND}/>;
  }

  const {
    primaryBackgroundStyle,
  } = movie;

  return <>
    <section className="movie-card movie-card--full" style={primaryBackgroundStyle}>

      <div className="movie-card__hero">
        <MovieBackground movie={movie}/>

        <h1 className="visually-hidden">WTW</h1>

        <header className="page-header movie-card__head">
          <Logo/>

          <UserBlock/>
        </header>

        <div className="movie-card__wrap">
          <MovieDescription movie={movie}>
            {loginStatus === OperationStatus.RESOLVED && (
              <Link className="btn movie-card__button" to={generatePath(MainPath.ADD_REVIEW, movie)}>
                Add review
              </Link>
            )}
          </MovieDescription>
        </div>
      </div>

      <div className="movie-card__wrap movie-card__translate-top">
        <MovieInfo movie={movie} posterSize={PosterSize.BIG}/>
      </div>
    </section>

    <div className="page-content">
      {alikeMovies.length > 0 && (
        <section className="catalog catalog--like-this">
          <h2 className="catalog__title">More like this</h2>
          <MoviesList movies={alikeMovies}/>
        </section>
      )}

      <Footer/>
    </div>
  </>;
};

FullMovieCard.propTypes = {
  loginStatus: operationStatusType,
  moviesStatus: operationStatusType,
  movie: movieShape,
  alikeMovies: OriginalMoviesList.propTypes.movies,
};

const mapStateToProps = (state, {movieId}) => {
  return {
    loginStatus: selectLoginStatus(state),
    moviesStatus: selectMoviesStatus(state),
    movie: selectMovieById(state, movieId),
    alikeMovies: selectAlikeMovies(state, movieId),
  };
};

export {FullMovieCard};
export default connect(mapStateToProps)(FullMovieCard);
