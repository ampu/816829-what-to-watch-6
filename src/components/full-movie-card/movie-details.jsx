import React from 'react';
import PropTypes from 'prop-types';

import {formatDuration} from '../../utils/date-util';

import {movieType} from '../../typings/movie-type';

import './movie-details.css';

const MovieDetails = ({movie = {}} = {}) => {
  const {
    director = ``,
    stars = [],
    durationSeconds = 0,
    genre = ``,
    year = 0,
  } = movie;

  return (
    <div className="movie-card__text movie-card__row">
      <div className="movie-card__text-col">
        {director && (
          <p className="movie-card__details-item">
            <strong className="movie-card__details-name">Director</strong>
            <span className="movie-card__details-value">{director}</span>
          </p>
        )}
        {stars.length > 0 && (
          <p className="movie-card__details-item">
            <strong className="movie-card__details-name">Starring</strong>
            <span className="movie-card__details-value movie-card__details-value--stars">
              {stars.join(`,\n`)}
            </span>
          </p>
        )}
      </div>

      <div className="movie-card__text-col">
        {durationSeconds > 0 && (
          <p className="movie-card__details-item">
            <strong className="movie-card__details-name">Run Time</strong>
            <span className="movie-card__details-value">
              {formatDuration(durationSeconds, `H[h] m[m]`)}
            </span>
          </p>
        )}
        {genre && (
          <p className="movie-card__details-item">
            <strong className="movie-card__details-name">Genre</strong>
            <span className="movie-card__details-value">{genre}</span>
          </p>
        )}
        {year > 0 && (
          <p className="movie-card__details-item">
            <strong className="movie-card__details-name">Released</strong>
            <span className="movie-card__details-value">{year}</span>
          </p>
        )}
      </div>
    </div>
  );
};

MovieDetails.propTypes = {
  movie: PropTypes.shape({
    director: movieType.director,
    stars: movieType.stars,
    durationSeconds: movieType.durationSeconds,
    genre: movieType.genre,
    year: movieType.year,
  }),
};

export default MovieDetails;
