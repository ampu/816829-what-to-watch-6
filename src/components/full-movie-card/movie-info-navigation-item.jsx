import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';
import getClassName from 'classnames';

import {useMatchPath} from '../../hooks/use-match-path';

const MovieInfoNavigationItem = ({title, path}) => {

  const classMap = {
    [`movie-nav__item`]: true,
    [`movie-nav__item--active`]: useMatchPath(path),
  };

  return (
    <li className={getClassName(classMap)}>
      <Link className="movie-nav__link" to={path}>{title}</Link>
    </li>
  );
};

MovieInfoNavigationItem.propTypes = {
  title: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
};

export default MovieInfoNavigationItem;
