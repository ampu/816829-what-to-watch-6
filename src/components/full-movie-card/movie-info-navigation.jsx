import React from 'react';
import PropTypes from 'prop-types';
import {generatePath} from 'react-router-dom';

import {MoviePath} from '../../constants/paths';

import MovieInfoNavigationItem from './movie-info-navigation-item';

import {movieType} from '../../typings/movie-type';

const MovieInfoNavigation = ({movie}) => {
  return (
    <nav className="movie-nav movie-card__nav">
      <ul className="movie-nav__list">
        <MovieInfoNavigationItem title="Overview" path={generatePath(MoviePath.MOVIE_OVERVIEW, movie)}/>
        <MovieInfoNavigationItem title="Details" path={generatePath(MoviePath.MOVIE_DETAILS, movie)}/>
        <MovieInfoNavigationItem title="Reviews" path={generatePath(MoviePath.MOVIE_REVIEWS, movie)}/>
      </ul>
    </nav>
  );
};

MovieInfoNavigation.propTypes = {
  movie: PropTypes.shape({
    id: movieType.id,
  }).isRequired,
};

export default MovieInfoNavigation;
