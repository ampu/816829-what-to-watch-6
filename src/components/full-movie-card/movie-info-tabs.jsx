import React from 'react';

import {MoviePath} from '../../constants/paths';
import {useMatchPath} from '../../hooks/use-match-path';

import MovieOverview from './movie-overview';
import MovieDetails from './movie-details';
import MovieReviews from './movie-reviews';

import {movieShape} from '../../typings/movie-type';

const MovieInfoTabs = ({movie}) => {
  return <>
    {useMatchPath(MoviePath.MOVIE_OVERVIEW) && <MovieOverview movie={movie}/>}
    {useMatchPath(MoviePath.MOVIE_DETAILS) && <MovieDetails movie={movie}/>}
    {useMatchPath(MoviePath.MOVIE_REVIEWS) && <MovieReviews movie={movie}/>}
  </>;
};

MovieInfoTabs.propTypes = {
  movie: movieShape,
};

export default MovieInfoTabs;
