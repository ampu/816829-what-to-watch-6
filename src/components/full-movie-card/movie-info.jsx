import React from 'react';

import MoviePoster from '../movie-poster/movie-poster';
import MovieInfoNavigation from './movie-info-navigation';
import MovieInfoTabs from './movie-info-tabs';

import {movieShape} from '../../typings/movie-type';

const MovieInfo = ({movie = {}, posterSize} = {}) => {

  return (
    <div className="movie-card__info">
      <MoviePoster size={posterSize} movie={movie}/>

      <div className="movie-card__desc">
        <MovieInfoNavigation movie={movie}/>
        <MovieInfoTabs movie={movie}/>
      </div>
    </div>
  );
};

MovieInfo.propTypes = {
  movie: movieShape,
  posterSize: MoviePoster.propTypes.size,
};

export default MovieInfo;
