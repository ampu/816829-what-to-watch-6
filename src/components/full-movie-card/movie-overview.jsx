import React from 'react';
import PropTypes from 'prop-types';

import {getRatingLevel, formatScoresCount, formatRating} from '../../utils/movie-util';

import movieType from '../../typings/movie-type';

const MovieOverview = ({movie = {}} = {}) => {
  const {
    description = ``,
    director = ``,
    stars = ``,
    rating = 0,
    scoresCount = 0,
  } = movie;

  return <>
    {scoresCount > 0 && (
      <div className="movie-rating">
        <div className="movie-rating__score">{formatRating(rating)}</div>
        <p className="movie-rating__meta">
          <span className="movie-rating__level">{getRatingLevel(rating)}</span>
          <span className="movie-rating__count">{formatScoresCount(scoresCount)} ratings</span>
        </p>
      </div>
    )}

    <div className="movie-card__text">
      {description && <p>{description}</p>}

      {director && <p className="movie-card__director"><strong>Director: {director}</strong></p>}

      {stars && <p className="movie-card__starring"><strong>Starring: {stars.join(`, `)}</strong></p>}
    </div>
  </>;
};

MovieOverview.propTypes = {
  movie: PropTypes.shape({
    description: movieType.description,
    director: movieType.director,
    stars: movieType.stars,
    rating: movieType.rating,
    scoresCount: movieType.scoresCount,
  }),
};

export default MovieOverview;
