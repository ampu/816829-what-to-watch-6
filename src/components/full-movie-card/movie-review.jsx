import React from 'react';
import PropTypes from 'prop-types';

import {BORDER_BOTTOM_STYLE} from '../../constants/styles';
import {formatRating} from '../../utils/movie-util';
import {formatDate} from '../../utils/date-util';

import reviewType from '../../typings/review-type';

const MovieReview = ({review = {}}) => {
  const {
    user = {},
    text = ``,
    date = ``,
    rating = 0,
  } = review;

  return (
    <div className="review" style={BORDER_BOTTOM_STYLE}>
      <blockquote className="review__quote">
        {text && <p className="review__text">{text}</p>}

        <footer className="review__details">
          {user.name && <cite className="review__author">{user.name}</cite>}
          {date && (
            <time className="review__date" dateTime={formatDate(date, `YYYY-MM-DD`)}>
              {formatDate(date, `MMMM DD, YYYY`)}
            </time>
          )}
        </footer>
      </blockquote>

      {rating && <div className="review__rating">{formatRating(rating)}</div>}
    </div>
  );
};

MovieReview.propTypes = {
  review: PropTypes.shape(reviewType),
};

export default MovieReview;
