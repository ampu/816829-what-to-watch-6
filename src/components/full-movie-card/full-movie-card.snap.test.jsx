import React from 'react';
import {Router, generatePath} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import * as redux from 'react-redux';

import {render} from '@testing-library/react';

import {MainPath, MoviePath} from '../../constants/paths';

import {FullMovieCard} from './full-movie-card';
import Domain from '../../store/domain';
import OperationStatus from '../../constants/operation-status';
import {ALL_GENRES} from '../../constants/genre';
import configureStore from 'redux-mock-store';

const INITIAL_STATE = {
  [Domain.MOVIE]: {
    moviesStatus: OperationStatus.UNSET,
    movies: [],
    promoStatus: OperationStatus.UNSET,
    promoMovie: undefined,
  },
  [Domain.USER]: {
    loginStatus: OperationStatus.UNSET,
    user: undefined,
  },
  [Domain.GENRE]: {
    genre: ALL_GENRES,
  },
};

const thunkMock = () => (next) => (action) => {
  if (typeof action === `function`) {
    return Promise.resolve();
  }
  return next(action);
};

const initializeStoreWithState = configureStore([thunkMock]);

describe(`FullMovieCard component`, () => {
  it(`Should render according to the snapshot when movies request is pending`, () => {
    const {container} = render(<FullMovieCard moviesStatus={OperationStatus.PENDING} alikeMovies={[]}/>);
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot when movies request was rejected`, () => {
    const {container} = render(<FullMovieCard moviesStatus={OperationStatus.REJECTED} alikeMovies={[]}/>);
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot when no movie is provided`, () => {
    const {container} = render((
      <Router history={createMemoryHistory()}>
        <FullMovieCard alikeMovies={[]}/>
      </Router>
    ));
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot while at '/' url`, () => {
    const movie = {id: `fakeId`, poster: `fakePoster`};

    const {container} = render((
      <redux.Provider store={initializeStoreWithState(INITIAL_STATE)}>
        <Router history={createMemoryHistory()}>
          <FullMovieCard movie={movie} alikeMovies={[]}/>
        </Router>
      </redux.Provider>
    ));
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot when alike movies are provided`, () => {
    const movie = {id: `fakeId`, poster: `fakePoster`};

    const alikeMovies = [
      {id: `fakeId-1`, preview: `fakePoster`, videoPreview: `fakeVideoPreview`},
      {id: `fakeId-2`, preview: `fakePoster`, videoPreview: `fakeVideoPreview`},
    ];

    const {container} = render((
      <redux.Provider store={initializeStoreWithState(INITIAL_STATE)}>
        <Router history={createMemoryHistory()}>
          <FullMovieCard movie={movie} alikeMovies={alikeMovies}/>
        </Router>
      </redux.Provider>
    ));
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot while at '/films/:id' url`, () => {
    const movie = {id: `fakeId`, poster: `fakePoster`};

    const history = createMemoryHistory();
    history.push(generatePath(MainPath.MOVIE, movie));

    const {container} = render((
      <redux.Provider store={initializeStoreWithState(INITIAL_STATE)}>
        <Router history={history}>
          <FullMovieCard movie={movie} alikeMovies={[]}/>
        </Router>
      </redux.Provider>
    ));
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot while at '/films/:id/details' url`, () => {
    const movie = {id: `fakeId`, poster: `fakePoster`};

    const history = createMemoryHistory();
    history.push(generatePath(MoviePath.MOVIE_DETAILS, movie));

    const {container} = render((
      <redux.Provider store={initializeStoreWithState(INITIAL_STATE)}>
        <Router history={history}>
          <FullMovieCard movie={movie} alikeMovies={[]}/>
        </Router>
      </redux.Provider>
    ));
    expect(container).toMatchSnapshot();
  });
});
