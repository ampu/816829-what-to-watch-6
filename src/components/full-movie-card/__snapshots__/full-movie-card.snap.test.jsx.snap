// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullMovieCard component Should render according to the snapshot when alike movies are provided 1`] = `
<div>
  <section
    class="movie-card movie-card--full"
  >
    <div
      class="movie-card__hero"
    >
      
      <h1
        class="visually-hidden"
      >
        WTW
      </h1>
      <header
        class="page-header movie-card__head"
      >
        <div
          class="logo"
        >
          <a
            aria-current="page"
            class="logo__link logo__link--active"
            href="/"
          >
            <span
              class="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              class="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              class="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          class="user-block"
        />
      </header>
      <div
        class="movie-card__wrap"
      >
        <div
          class="movie-card__desc"
        >
          <h2
            class="movie-card__title"
          />
          <p
            class="movie-card__meta"
          >
            
            
          </p>
          <div
            class="movie-card__buttons"
          >
            <button
              class="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlink:href="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="movie-card__wrap movie-card__translate-top"
    >
      <div
        class="movie-card__info"
      >
        <div
          class="movie-card__poster movie-card__poster--big"
        >
          <img
            alt=" poster"
            height="327"
            src="fakePoster"
            width="218"
          />
        </div>
        <div
          class="movie-card__desc"
        >
          <nav
            class="movie-nav movie-card__nav"
          >
            <ul
              class="movie-nav__list"
            >
              <li
                class="movie-nav__item"
              >
                <a
                  class="movie-nav__link"
                  href="/films/fakeId"
                >
                  Overview
                </a>
              </li>
              <li
                class="movie-nav__item"
              >
                <a
                  class="movie-nav__link"
                  href="/films/fakeId/details"
                >
                  Details
                </a>
              </li>
              <li
                class="movie-nav__item"
              >
                <a
                  class="movie-nav__link"
                  href="/films/fakeId/reviews"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </section>
  <div
    class="page-content"
  >
    <section
      class="catalog catalog--like-this"
    >
      <h2
        class="catalog__title"
      >
        More like this
      </h2>
      <div
        class="catalog__movies-list"
      >
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            <img
              alt=""
              height="175"
              src="fakePoster"
              width="280"
            />
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/fakeId-1"
            />
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            <img
              alt=""
              height="175"
              src="fakePoster"
              width="280"
            />
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/fakeId-2"
            />
          </h3>
        </article>
      </div>
    </section>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          aria-current="page"
          class="logo__link logo__link--light logo__link--active"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;

exports[`FullMovieCard component Should render according to the snapshot when movies request is pending 1`] = `
<div>
  <div
    class="container container--centered"
  >
    <div
      class="spinner-loading"
    >
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;

exports[`FullMovieCard component Should render according to the snapshot when movies request was rejected 1`] = `
<div>
  <div
    class="container container--centered"
  >
    <div
      class="maintenance"
    >
      Maintenance, sorry...
    </div>
  </div>
</div>
`;

exports[`FullMovieCard component Should render according to the snapshot when no movie is provided 1`] = `<div />`;

exports[`FullMovieCard component Should render according to the snapshot while at '/' url 1`] = `
<div>
  <section
    class="movie-card movie-card--full"
  >
    <div
      class="movie-card__hero"
    >
      
      <h1
        class="visually-hidden"
      >
        WTW
      </h1>
      <header
        class="page-header movie-card__head"
      >
        <div
          class="logo"
        >
          <a
            aria-current="page"
            class="logo__link logo__link--active"
            href="/"
          >
            <span
              class="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              class="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              class="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          class="user-block"
        />
      </header>
      <div
        class="movie-card__wrap"
      >
        <div
          class="movie-card__desc"
        >
          <h2
            class="movie-card__title"
          />
          <p
            class="movie-card__meta"
          >
            
            
          </p>
          <div
            class="movie-card__buttons"
          >
            <button
              class="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlink:href="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="movie-card__wrap movie-card__translate-top"
    >
      <div
        class="movie-card__info"
      >
        <div
          class="movie-card__poster movie-card__poster--big"
        >
          <img
            alt=" poster"
            height="327"
            src="fakePoster"
            width="218"
          />
        </div>
        <div
          class="movie-card__desc"
        >
          <nav
            class="movie-nav movie-card__nav"
          >
            <ul
              class="movie-nav__list"
            >
              <li
                class="movie-nav__item"
              >
                <a
                  class="movie-nav__link"
                  href="/films/fakeId"
                >
                  Overview
                </a>
              </li>
              <li
                class="movie-nav__item"
              >
                <a
                  class="movie-nav__link"
                  href="/films/fakeId/details"
                >
                  Details
                </a>
              </li>
              <li
                class="movie-nav__item"
              >
                <a
                  class="movie-nav__link"
                  href="/films/fakeId/reviews"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </section>
  <div
    class="page-content"
  >
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          aria-current="page"
          class="logo__link logo__link--light logo__link--active"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;

exports[`FullMovieCard component Should render according to the snapshot while at '/films/:id' url 1`] = `
<div>
  <section
    class="movie-card movie-card--full"
  >
    <div
      class="movie-card__hero"
    >
      
      <h1
        class="visually-hidden"
      >
        WTW
      </h1>
      <header
        class="page-header movie-card__head"
      >
        <div
          class="logo"
        >
          <a
            class="logo__link"
            href="/"
          >
            <span
              class="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              class="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              class="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          class="user-block"
        />
      </header>
      <div
        class="movie-card__wrap"
      >
        <div
          class="movie-card__desc"
        >
          <h2
            class="movie-card__title"
          />
          <p
            class="movie-card__meta"
          >
            
            
          </p>
          <div
            class="movie-card__buttons"
          >
            <button
              class="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlink:href="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="movie-card__wrap movie-card__translate-top"
    >
      <div
        class="movie-card__info"
      >
        <div
          class="movie-card__poster movie-card__poster--big"
        >
          <img
            alt=" poster"
            height="327"
            src="fakePoster"
            width="218"
          />
        </div>
        <div
          class="movie-card__desc"
        >
          <nav
            class="movie-nav movie-card__nav"
          >
            <ul
              class="movie-nav__list"
            >
              <li
                class="movie-nav__item movie-nav__item--active"
              >
                <a
                  class="movie-nav__link"
                  href="/films/fakeId"
                >
                  Overview
                </a>
              </li>
              <li
                class="movie-nav__item"
              >
                <a
                  class="movie-nav__link"
                  href="/films/fakeId/details"
                >
                  Details
                </a>
              </li>
              <li
                class="movie-nav__item"
              >
                <a
                  class="movie-nav__link"
                  href="/films/fakeId/reviews"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            class="movie-card__text"
          >
            
            
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    class="page-content"
  >
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link logo__link--light"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;

exports[`FullMovieCard component Should render according to the snapshot while at '/films/:id/details' url 1`] = `
<div>
  <section
    class="movie-card movie-card--full"
  >
    <div
      class="movie-card__hero"
    >
      
      <h1
        class="visually-hidden"
      >
        WTW
      </h1>
      <header
        class="page-header movie-card__head"
      >
        <div
          class="logo"
        >
          <a
            class="logo__link"
            href="/"
          >
            <span
              class="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              class="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              class="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          class="user-block"
        />
      </header>
      <div
        class="movie-card__wrap"
      >
        <div
          class="movie-card__desc"
        >
          <h2
            class="movie-card__title"
          />
          <p
            class="movie-card__meta"
          >
            
            
          </p>
          <div
            class="movie-card__buttons"
          >
            <button
              class="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlink:href="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="movie-card__wrap movie-card__translate-top"
    >
      <div
        class="movie-card__info"
      >
        <div
          class="movie-card__poster movie-card__poster--big"
        >
          <img
            alt=" poster"
            height="327"
            src="fakePoster"
            width="218"
          />
        </div>
        <div
          class="movie-card__desc"
        >
          <nav
            class="movie-nav movie-card__nav"
          >
            <ul
              class="movie-nav__list"
            >
              <li
                class="movie-nav__item"
              >
                <a
                  class="movie-nav__link"
                  href="/films/fakeId"
                >
                  Overview
                </a>
              </li>
              <li
                class="movie-nav__item movie-nav__item--active"
              >
                <a
                  class="movie-nav__link"
                  href="/films/fakeId/details"
                >
                  Details
                </a>
              </li>
              <li
                class="movie-nav__item"
              >
                <a
                  class="movie-nav__link"
                  href="/films/fakeId/reviews"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            class="movie-card__text movie-card__row"
          >
            <div
              class="movie-card__text-col"
            >
              
            </div>
            <div
              class="movie-card__text-col"
            >
              
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    class="page-content"
  >
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link logo__link--light"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
